src/flow/ext_multiroute_flow.jl

```diff
@@ -0,0 +1,231 @@
+  auxpoints[λ + 1 - s2] = (r2, f2) # Add right initial auxiliary point
+
+  # Loop if the slopes are distinct by at least 2
+  if s1 > s2 + 1
+    queue = [((f1, s1, r1), (f2, s2, r2))]
+
+    while !isempty(queue)
+      # Computes an intersection (middle) with a new associated slope
+      (f1, s1, r1), (f2, s2, r2) = pop!(queue)
+      r, expectedflow = intersection(r1, f1, s1, r2, f2, s2)
+      f, F, cut = maximum_flow(flow_graph, source, target, capacity_matrix,
+                  algorithm = BoykovKolmogorovAlgorithm(), restriction = r)
+      s = slope(flow_graph, capacity_matrix, max(cut, 1), r) # current slope
+      auxpoints[λ + 1 - s] = (r, f)
+      # If the flow at the intersection (middle) is as expected
+      if false # expectedflow ≈ f # approximatively equal (enforced by floating precision)
+        for k in (s1 - 1):(s2 + 1)
+          auxpoints[λ + 1 - k] = (r, f) # add all points between left and right
+        end
+      else
+        # if the slope difference between (middle) and left is at least 2
+        # push (left),(middle)
+        if s1 > s + 1 && (r2, f2) ≉ (r, f)
+          q = (f1, s1, r1), (f, s, r)
+          push!(queue, q)
+        end
+        # if the slope difference between (middle) and right is at least 2
+        # push (middle),(right)
+        if s > s2 + 1 && (r1, f1) ≉ (r, f)
+          q = (f, s, r), (f2, s2, r2)
+          push!(queue, q)
+        end
+      end
+    end
+  end
+  return auxpoints
+end
+
+"""
+Calculates the breaking of the restricted max-flow from a set of auxiliary points.
+Requires arguments:
+- flow_graph::DiGraph                    # the input graph
+- source::Int                            # the source vertex
+- target::Int                            # the target vertex
+- capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+"""
+
+function breakingPoints{T<:AbstractFloat}(
+  flow_graph::DiGraph,                   # the input graph
+  source::Int,                           # the source vertex
+  target::Int,                           # the target vertex
+  capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+  )
+  auxpoints = auxiliaryPoints(flow_graph, source, target, capacity_matrix)
+  λ = length(auxpoints) - 1
+  left_index = 1
+  breakingpoints = Vector{Tuple{T,T,Int}}()
+
+  for (id,point) in enumerate(auxpoints)
+    if id == 1
+      push!(breakingpoints, (0., 0., λ))
+    else
+      pleft = breakingpoints[left_index]
+      if point[1] != 0
+        x, y = intersection(pleft[1], pleft[2], pleft[3], point[1], point[2], λ + 1 - id)
+        push!(breakingpoints,(x, y, λ + 1 - id))
+        left_index += 1
+      end
+    end
+  end
+  return breakingpoints
+end
+
+"""
+Function to get the nonzero min and max function of a Matrix
+Requires argument:
+- capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+"""
+
+# Function to get the nonzero min and max function of a Matrix
```

Comment:
We could actually implement those three function within an optional interval of accepted values and add them to the Base.array (or whatever corresponds to the Array part of the core library).

What do you think?